src/Microsoft.ML.TorchSharp/NasBert/TextClassificationTrainer.cs

```diff
@@ -328,13 +328,17 @@ public Trainer(TextClassificationTrainer parent, IChannel ch)
                 var vocabulary = Tokenizer.Vocabulary;
                 vocabulary.AddMaskSymbol();
 
+                // Get row count and figure out num of unique labels
+                var rowCount = GetRowCountAndSetLabelCount(input);
+
                 // Initialize the model and load pre-trained weights
                 Model = new TextClassificationModel(_parent._options, vocabulary, _parent._options.NumberOfClasses);
                 Model.GetEncoder().load(GetModelPath());
                 Model.train();
 
                 // Figure out if we are running on GPU or CPU
                 Device = ((IHostEnvironmentInternal)_parent._host).GpuDeviceId != null && cuda.is_available() ? CUDA : CPU;
+                Contracts.Assert(Device == CPU || (((IHostEnvironmentInternal)_parent._host).FallbackToCpu != false || Device != CPU), "Fallback to CPU is false but no GPU detected");
```

Comment:
>Device != CPU [](http://example.com/codeflow?start=119&length=13)

This part is not really needed

Reply:
Why is that part not needed? Basically what I was trying to do is take this logic:
`if fallbackToCpu == false and Device == CPU then throw and error`
and just invert it.

Reply:
Though I may still not quite have it right. Let me test again. We don't have a GPU machine in the CI so all the testing for this is manual.

Reply:
if you have Device == CPU then the whole check `(((IHostEnvironmentInternal)_parent._host).FallbackToCpu != false || Device != CPU)` will not be tested anyway.

Reply:
Changes have been pushed.